deployment/constructs/dynamodb-table.ts

```diff
@@ -23,21 +23,20 @@ export interface DynamoDBTableIdentifiers {
 }
 
 export class DynamoDBTable extends Construct {
-  public readonly table: dynamodb.Table;
+  public readonly table: dynamodb.TableV2;
   public readonly identifiers: DynamoDBTableIdentifiers;
 
   constructor(scope: Construct, id: string, props: DynamoDBTableProps) {
     super(scope, id);
 
     const tableName = `${props.stage}-${props.name}`;
-    this.table = new dynamodb.Table(this, "Table", {
+    this.table = new dynamodb.TableV2(this, "Table", {
```

Comment:
Switching to v2 is recommended esp if we go this route where the tables change to a new table with a new name. Advantage according to AWS is that with V2 everything is a "Global Table" meaning you can have cross region replicas but you can't take a v1 table and do that so they recommend making all tables as global tables now.